packages/services/src/infrastructure/api/binance/binance-api.client.ts

```diff
@@ -0,0 +1,43 @@
+import axios from 'axios';
+
+import { CryptoCurrency } from '@leather.io/models';
+
+import { CacheService } from '../../cache/cache.service';
+import { CacheTimeMs } from '../../cache/cache.utils';
+import { readBinanceMarketDataPrice } from './binance-api.utils';
+
+export interface BinanceMarketData {
+  symbol: string;
+  price: string;
+}
+
+export interface BinanceApiClient {
+  fetchMarketData(currency: CryptoCurrency, signal?: AbortSignal): Promise<BinanceMarketData>;
+  fetchMarketDataPrice(currency: CryptoCurrency, signal?: AbortSignal): Promise<number | undefined>;
+}
+
+export function createBinanceApiClient(cacheService: CacheService) {
+  async function fetchMarketDataPrice(currency: CryptoCurrency, signal?: AbortSignal) {
+    const marketData = await fetchMarketData(currency, signal);
+    return readBinanceMarketDataPrice(marketData);
```

Comment:
I don't think that should be the responsibility of client, but it should be an option offered via an associated client utility. 

If we tried to convert all client response to internal type by default, it would require a lot of assumptions about how it should be used - i.e. formatting, what data gets included, what data gets excluded, etc... By being faithful to the API return type, we separate concerns and allow consuming services to decide how the response data should be used. 

For things like converting a particular API response to an internal type like `Money`, I think that would be a great example of a utility that can live alongside the client and used selectively in services when required.